### PR TITLE
Fix leveldb file increase

### PR DESCRIPTION
**Pull request by Ashuaidehao**
Release file handles.Relate to https://github.com/neo-project/neo-plugins/pull/139

### CODE REVIEW

File: neo/SmartContract/ApplicationEngine.cs [hunk 1]
```diff
@@ -162,6 +162,7 @@ protected override bool PreExecuteInstruction()
             ApplicationEngine engine = new ApplicationEngine(TriggerType.Application, container, snapshot, extraGAS, testMode);
             engine.LoadScript(script);
             engine.Execute();
+            engine.Dispose();
```
Comment (erikzhang) [inline neo/SmartContract/ApplicationEngine.cs]: The `engine` should be disposed by the caller.

Comment (Ashuaidehao) [reply neo/SmartContract/ApplicationEngine.cs]: Seems most caller just call it one time without release, we need a way to fix it once for all.
(https://github.com/neo-project/neo-plugins/blob/29fee59944fb4ae3a9e8d644c3d334fd3df5b474/RpcNep5Tracker/RpcNep5Tracker.cs#L168)
Changing it in neo.dll could eliminate the hidden trouble.
Ortherwise, we need check all plugins who call it and refactor.

Comment (erikzhang) [reply neo/SmartContract/ApplicationEngine.cs]: You need to access the object, why dispose it before using?

Comment (Ashuaidehao) [reply neo/SmartContract/ApplicationEngine.cs]: For most conditions, we just use the method to get the run results not engine itself. 
Sometime the smartcontract may invoke needing more gas than the gas limit. If this happen when opening leveldb iterators, the engine interrupts and these iterators cannot close normally. Unless call engine "dispose" manually.
Seem gas limit interrupts may happen in many unpredictable situations, fix the hidden leak trouble here could nip in the bud.

**Comment by erikzhang** [comment]
See #1333

**Pull request closed without merging**